rust-server/src/main.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::Result;
+use futures::StreamExt;
+use libp2p::{
+    core::{muxing::StreamMuxerBox},
+    gossipsub,
+    identity,
+    kad::record::store::{RecordStore, MemoryStore},
+    kad::{GetClosestPeersError, Kademlia, KademliaConfig, KademliaEvent, QueryResult},
+    PeerId,
+    swarm::{NetworkBehaviour, Swarm, SwarmBuilder},
+    Transport,
+    webrtc,
+};
+use rand::thread_rng;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::time::Duration;
+
+// TODO: replace with our private bootstrap node
+const BOOTNODES: [&str; 1] = [
+    "QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN",
+    // "QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
+    // "QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
+    // "QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
+];
+
+/// An example WebRTC server that will accept connections
+#[tokio::main]
+async fn main() -> Result<()> {
+    let mut swarm = create_swarm::<MemoryStore>()?;
+
+    swarm.listen_on("/ip4/127.0.0.1/udp/0/webrtc".parse()?)?;
+
+    loop {
+        let event = swarm.next().await.unwrap();
+        eprintln!("New event: {event:?}")
+    }
+}
+
+#[derive(NetworkBehaviour)]
+struct Behaviour<T: RecordStore + std::marker::Send + 'static> {
+    gossipsub: gossipsub::Behaviour,
+    kademlia: Kademlia<T>,
```

Comment:
Consider adding the `libp2p-ping` protocol here. Useful in general, but in particular to debug with a **very** simple protocol.

See also https://github.com/libp2p/rust-libp2p/tree/master/examples/ping-example

Reply:
added ping and keep_alive